Core/include/Acts/Surfaces/BoundaryTolerance.hpp

```diff
@@ -0,0 +1,154 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2016-2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+
+#include <cmath>
+#include <iterator>
+#include <optional>
+#include <variant>
+#include <vector>
+
+namespace Acts {
+
+/// Captures different types of boundary tolerances
+/// - Infinite: infinite tolerance i.e. no boundary check
+/// - None: no tolerance i.e. exact boundary check
+/// - AbsoluteBound: absolute tolerance in bound coordinates
+/// - AbsoluteCartesian: absolute tolerance in Cartesian coordinates
+/// - AbsoluteEuclidean: absolute tolerance in Euclidean distance
+/// - Chi2Bound: chi2 tolerance in bound coordinates
+class BoundaryTolerance {
```

Comment:
I think we might need some more documentation about what these cases are, possibly with pictures so make it clear. At least the euclidean distance should be described in a bit more detail here.

Reply:
True documentation is lacking quite a big even before. Do you think this should be done directly in this PR or it could potentially wait for a follow-up?

Reply:
I tried to improve the doc